ecosystem/indexer-grpc/indexer-grpc-integration-test-transaction-generator/src/move_executor.rs

```diff
@@ -0,0 +1,305 @@
+            if !output.status.success() {
+                return Err(anyhow::anyhow!(
+                    "Failed to publish Move module: {:?}",
+                    output
+                ));
+            }
+            Ok(None)
+        } else {
+            // Compile the Move script first.
+            let mut compile_cmd = tokio::process::Command::new(APTOS_CLI_BINARY_NAME);
+            compile_cmd.arg("move").arg("compile");
+            let output = compile_cmd
+                .output()
+                .await
+                .context(format!("Failed to run script {:?}", move_module_folder))?;
+            if !output.status.success() {
+                return Err(anyhow::anyhow!(
+                    "Failed to compile Move script: {:?}",
+                    output
+                ));
+            }
+            // Parse the Move.toml file to get the module name.
+            let move_toml_path = move_module_folder.join("Move.toml");
+            let move_toml_content = tokio::fs::read_to_string(&move_toml_path).await?;
+            let move_config: MoveConfig = toml::from_str(&move_toml_content).context(format!(
+                "Failed to parse Move.toml file at {:?}",
+                move_toml_path
+            ))?;
+            let package_name = move_config.package.name;
+
+            // Run the compiled Move Script.
+            let mut run_cmd = tokio::process::Command::new(APTOS_CLI_BINARY_NAME);
+            run_cmd
+                .arg("move")
+                .arg("run-script")
+                .arg("--compiled-script-path")
+                .arg(format!("build/{}/bytecode_scripts/main.mv", package_name))
+                .arg("--assume-yes");
+            match run_cmd.output().await {
+                Ok(output) => {
+                    if !output.status.success() {
+                        return Err(anyhow::anyhow!("Failed to run Move script: {:?}", output));
+                    }
+                    let aptos_cli_output: AptosCliOutput =
+                        match serde_json::from_slice(&output.stdout)
+                            .context("Failed to parse aptos output.")
+                        {
+                            Ok(output) => output,
+                            Err(e) => {
+                                // This means the
+                                return Err(anyhow::anyhow!(
+                                    "Failed to parse aptos output: {:?}",
+                                    e
+                                ));
+                            },
+                        };
+                    Ok(Some(aptos_cli_output.result.version))
+                },
+                Err(e) => Err(anyhow::anyhow!("Failed to run Move script: {:?}", e)),
+            }
+        }
+    }
+
+    // Internal methods.
+
+    /// Wait for the node to be ready.
+    async fn wait_if_not_ready(&self) -> anyhow::Result<()> {
```

Comment:
I feel like this is really reinventing the wheel, localnet already does lots of what we want, such as various configuration for the node, running the node in a way that works for localnet (altering the config), waiting for it to come up, etc.